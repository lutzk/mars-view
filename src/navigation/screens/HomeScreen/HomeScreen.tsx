import * as React from 'react';
import { Text, View } from 'react-native';

import strings from '../../../strings/strings.json';

const HomeScreen: React.FC = () => {
  return (
    <View style={{ flex: 1 }}>
      <Text>{strings.home.headline}</Text>
    </View>
  );
};

export { HomeScreen };
