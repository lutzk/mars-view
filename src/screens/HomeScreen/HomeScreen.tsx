import * as React from 'react';
import { Text, View } from 'react-native';

import strings from '@strings/strings.json';

const HomeScreen = () => {
  return (
    <View style={{ flex: 1 }}>
      <Text>{strings.home.headline}</Text>

      {/* remove the line and json import breaks WTF xD */}
      <Text></Text>
    </View>
  );
};

export { HomeScreen };
