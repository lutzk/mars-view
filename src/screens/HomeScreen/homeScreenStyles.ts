import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flexGrow: 1,
    padding: 20,
  },
});

export { styles };
