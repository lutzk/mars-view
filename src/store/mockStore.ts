import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { RootState } from './store';
import { RoverNames } from './features/rovers/roverSlice';
import { MissionStatus } from '@api/roverApi';

const middlewares = [thunk];
const createMockStore = configureMockStore<RootState>(middlewares);
const mockStore = createMockStore({
  _persist: {
    version: -1,
    rehydrated: true,
  },
  rovers: {
    rovers: [],
    roversManifests: {
      [RoverNames.Spirit]: {
        name: RoverNames.Spirit,
        status: 'complete' as MissionStatus,
        landing_date: '2004-01-04',
        launch_date: '2003-06-10',
        max_sol: 2208,
        max_date: '2010-03-21',
        total_photos: 124550,
        wikiUrl: 'https://en.wikipedia.org/wiki/Spirit_(rover)',
      },
      [RoverNames.Curiosity]: {
        name: RoverNames.Curiosity,
        status: 'active' as MissionStatus,
        landing_date: '2004-01-04',
        launch_date: '2003-06-10',
        max_sol: 2208,
        max_date: '2010-03-21',
        total_photos: 124550,
        wikiUrl: 'https://en.wikipedia.org/wiki/Curiosity_(rover)',
      },
      [RoverNames.Opportunity]: {
        name: RoverNames.Opportunity,
        status: 'complete' as MissionStatus,
        landing_date: '2004-01-04',
        launch_date: '2003-06-10',
        max_sol: 2208,
        max_date: '2010-03-21',
        total_photos: 124550,
        wikiUrl: 'https://en.wikipedia.org/wiki/Opportunity_(rover)',
      },
    },
  },
});

export { mockStore };
