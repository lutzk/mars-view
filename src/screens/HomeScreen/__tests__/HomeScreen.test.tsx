import 'react-native';
import React from 'react';

// Note: import explicitly to use the types shiped with jest.
import { it, expect } from '@jest/globals';

// Note: test renderer must be required after react-native.
import { render } from '@testing-library/react-native';

import { HomeScreen } from '../HomeScreen';
import { Provider } from 'react-redux';
import { mockStore } from '@store/mockStore';

it('renders correctly', () => {
  const app = render(
    <Provider store={mockStore}>
      <HomeScreen />
    </Provider>,
  ).toJSON();

  expect(app).toMatchSnapshot();
});
