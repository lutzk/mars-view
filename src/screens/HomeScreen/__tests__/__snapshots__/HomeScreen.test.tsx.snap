// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "flexGrow": 1,
      "padding": 20,
    }
  }
>
  <View>
    <Text>
      welcome
    </Text>
    <View
      style={
        {
          "backgroundColor": "#f6c2ac",
          "borderColor": "#D84315",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 0,
          "justifyContent": "space-between",
          "marginBottom": 10,
          "padding": 20,
        }
      }
    >
      <View>
        <Text
          style={
            {
              "color": "#D84315",
              "fontSize": 20,
              "fontWeight": "normal",
            }
          }
          testID="cardHeader"
        >
          spirit
        </Text>
        <Text>
          landing date:
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2004-01-04
          </Text>
        </Text>
        <Text>
          status: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            complete
          </Text>
        </Text>
        <Text>
          launch date: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2003-06-10
          </Text>
        </Text>
        <Text>
          max date: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2010-03-21
          </Text>
        </Text>
        <Text>
          max sol: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2208
          </Text>
        </Text>
        <Text>
          total photos:
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            124550
          </Text>
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#D84315",
                "fontSize": 16,
                "fontWeight": "normal",
                "textDecorationColor": "#D84315",
                "textDecorationLine": "underline",
              }
            }
          >
            spirit
             on wikipedia
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#f6c2ac",
          "borderColor": "#D84315",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 0,
          "justifyContent": "space-between",
          "marginBottom": 10,
          "padding": 20,
        }
      }
    >
      <View>
        <Text
          style={
            {
              "color": "#D84315",
              "fontSize": 20,
              "fontWeight": "normal",
            }
          }
          testID="cardHeader"
        >
          opportunity
        </Text>
        <Text>
          landing date:
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2004-01-04
          </Text>
        </Text>
        <Text>
          status: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            complete
          </Text>
        </Text>
        <Text>
          launch date: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2003-06-10
          </Text>
        </Text>
        <Text>
          max date: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2010-03-21
          </Text>
        </Text>
        <Text>
          max sol: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2208
          </Text>
        </Text>
        <Text>
          total photos:
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            124550
          </Text>
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#D84315",
                "fontSize": 16,
                "fontWeight": "normal",
                "textDecorationColor": "#D84315",
                "textDecorationLine": "underline",
              }
            }
          >
            opportunity
             on wikipedia
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "backgroundColor": "#f6c2ac",
          "borderColor": "#D84315",
          "borderRadius": 5,
          "borderWidth": 1,
          "flex": 1,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 0,
          "justifyContent": "space-between",
          "marginBottom": 10,
          "padding": 20,
        }
      }
    >
      <View>
        <Text
          style={
            {
              "color": "#D84315",
              "fontSize": 20,
              "fontWeight": "normal",
            }
          }
          testID="cardHeader"
        >
          curiosity
        </Text>
        <Text>
          landing date:
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2004-01-04
          </Text>
        </Text>
        <Text>
          status: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            active
          </Text>
        </Text>
        <Text>
          launch date: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2003-06-10
          </Text>
        </Text>
        <Text>
          max date: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2010-03-21
          </Text>
        </Text>
        <Text>
          max sol: 
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            2208
          </Text>
        </Text>
        <Text>
          total photos:
          <Text
            style={
              {
                "color": "#212121",
                "fontSize": 16,
                "fontWeight": "bold",
              }
            }
          >
            124550
          </Text>
        </Text>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "color": "#D84315",
                "fontSize": 16,
                "fontWeight": "normal",
                "textDecorationColor": "#D84315",
                "textDecorationLine": "underline",
              }
            }
          >
            curiosity
             on wikipedia
          </Text>
        </View>
      </View>
    </View>
    <Text />
  </View>
</RCTScrollView>
`;
